packages/core/admin/admin/src/services/users.ts

```diff
@@ -0,0 +1,92 @@
+import { FindRole, FindRoles } from '../../../shared/contracts/roles';
+import * as Users from '../../../shared/contracts/user';
+
+import { adminApi } from './api';
+
+const usersService = adminApi.injectEndpoints({
+  endpoints: (builder) => ({
+    /**
+     * users
+     */
+    updateUser: builder.mutation<
+      Users.Update.Response['data'],
+      Omit<Users.Update.Request['body'] & Users.Update.Params, 'blocked'>
+    >({
+      query: ({ id, ...body }) => ({
+        url: `/admin/users/${id}`,
+        method: 'PUT',
+        data: body,
+      }),
+      invalidatesTags: (_res, _err, { id }) => [{ type: 'User', id }],
+    }),
+    getUsers: builder.query<
+      {
+        users: Users.FindAll.Response['data']['results'];
+        pagination: Users.FindAll.Response['data']['pagination'] | null;
+      },
+      GetUsersParams | void
+    >({
+      query: ({ id, ...params } = {}) => ({
+        url: `/admin/users/${id ?? ''}`,
+        method: 'GET',
+        config: {
+          params,
+        },
+      }),
```

Comment:
I'm just wondering, there is a reason to make just one `getUsers` that handle both cases (findMany & findOne) instead of create two `getUsers` & `getUser`, we did it in this way for content-releases 🤔 

It's something more personal I think it's fine!

Reply:
When we were using react-query we found it easier to use one hook for a few of these bits and then we just normalised it to always be an array, I don't think there's a huge benefit to our way it would just require more refactoring work! I suppose it's more powerful when we share it across the code base e.g. if you're fetching users in content releases it makes more sense to use the hook from the admin imo 🤓